projects/can_transceiver/src/can_transceiver_ros_intf.cpp

```diff
@@ -44,12 +61,120 @@ class CanTransceiverIntf : public rclcpp::Node
+ */
+class CanSimIntf : public rclcpp::Node
+{
+public:
+    CanSimIntf()  //Our node which subs to topics
+    : Node("CanSimIntf")
+    {
+        // Subscriber
+        // There is no timer because subscriber will respond to whatever data is published to the topic /Simulator
+        // Topic: mock_gps
+        subscriptionGPS_ = this->create_subscription<custom_interfaces::msg::GPS>(
+          MOCK_GPS_TOPIC, QUEUE_SIZE, std::bind(&CanSimIntf::gps_callback, this, std::placeholders::_1));
+
+        // Topic: mock_wind_sensors
+        subscriptionWindSensor_ = this->create_subscription<custom_interfaces::msg::WindSensors>(
+          MOCK_WIND_SENSORS_TOPIC, QUEUE_SIZE, std::bind(&CanSimIntf::wind_callback, this, std::placeholders::_1));
+
+        // Publisher
+        publisherWindSensors_ =
+          this->create_publisher<custom_interfaces::msg::WindSensors>(WIND_SENSORS_TOPIC, QUEUE_SIZE);
+        publisherWindSensor_ =
+          this->create_publisher<custom_interfaces::msg::WindSensor>(FILTERED_WIND_SENSOR_TOPIC, QUEUE_SIZE);
+        publisherGPS_ = this->create_publisher<custom_interfaces::msg::GPS>(GPS_TOPIC, QUEUE_SIZE);
+        publisherGenericSensors_ =
+          this->create_publisher<custom_interfaces::msg::GenericSensors>(DATA_SENSORS_TOPIC, QUEUE_SIZE);
+        publisherBatteries_ = this->create_publisher<custom_interfaces::msg::Batteries>(BATTERIES_TOPIC, QUEUE_SIZE);
+        // Timer with 500ms delay
+        timer_ = this->create_wall_timer(500ms, std::bind(&CanSimIntf::timer_callback, this));
+    }
+
+private:
+    // Subscriber
+    void gps_callback(const custom_interfaces::msg::GPS::SharedPtr msg) const {}
+    void wind_callback(const custom_interfaces::msg::WindSensors::SharedPtr msg) const {}
+
+    // Publisher
+    void timer_callback()
+    {
+        // ** GPS **
+        custom_interfaces::msg::GPS gpsPubData;
+        gpsPubData.heading.heading   = PLACEHOLDER_VALUE;
+        gpsPubData.speed.speed       = PLACEHOLDER_VALUE;
+        gpsPubData.lat_lon.latitude  = PLACEHOLDER_VALUE;
+        gpsPubData.lat_lon.longitude = PLACEHOLDER_VALUE;
+        // Publishes msg
+        publisherGPS_->publish(gpsPubData);
+
+        // ** Wind Sensor (WSensor)**
+        custom_interfaces::msg::WindSensor WSensorData;
+        WSensorData.speed.speed = PLACEHOLDER_VALUE;
+        WSensorData.direction   = PLACEHOLDER_VALUE;
+        // Publishes msg
+        publisherWindSensor_->publish(WSensorData);
+
+        // ** Wind Sensors (WSensors)**
+        custom_interfaces::msg::WindSensors WSensorsData;
+        WSensorsData.wind_sensors[0].direction   = PLACEHOLDER_VALUE;
+        WSensorsData.wind_sensors[0].speed.speed = PLACEHOLDER_VALUE;
+        // Publishes msg
+        publisherWindSensors_->publish(WSensorsData);
+
+        // ** Batteries **
+        custom_interfaces::msg::Batteries WBatteriesData;
+        WBatteriesData.batteries[0].voltage = PLACEHOLDER_VALUE;
```

Comment:
These are statically sized arrays. You should fill all entries. For example:
```
for (auto &sensor : WSensors.wind_sensors) {
    sensor.direction = PLACEHOLDER_VALUE;
    sensor.speed.speed = PLACEHOLDER_VALUE;
}
```

Reply:
replaced with for loop